sdk/cdsclient/client_queue.go

```diff
@@ -50,101 +52,86 @@ func shrinkQueue(queue *sdk.WorkflowQueue, nbJobsToKeep int) time.Time {
+		for ctx.Err() == nil {
+			if err := c.RequestSSEGet(ctx, "/events", chanSSEvt); err != nil {
+				log.Println(err)
+			}
+			time.Sleep(5 * time.Second)
+		}
+	})
 
 	for {
 		select {
 		case <-ctx.Done():
-			jobsTicker.Stop()
 			pbjobsTicker.Stop()
-			oldJobsTicker.Stop()
+			jobsTicker.Stop()
 			if jobs != nil {
 				close(jobs)
 			}
 			if pbjobs != nil {
 				close(pbjobs)
 			}
 			return ctx.Err()
-		case <-oldJobsTicker.C:
-			if c.config.Verbose {
-				fmt.Println("oldJobsTicker")
-			}
-
-			if jobs != nil {
-				queue := sdk.WorkflowQueue{}
-				if _, err := c.GetJSON("/queue/workflows", &queue); err != nil {
-					errs <- sdk.WrapError(err, "Unable to load old jobs")
-				}
+		case evt := <-chanSSEvt:
+			content, _ := ioutil.ReadAll(evt.Data)
+
+			var apiEvent sdk.Event
+			_ = json.Unmarshal(content, &apiEvent) // ignore errors
+			// filter only EventRunWorkflowJob
+			if apiEvent.EventType == "sdk.EventRunWorkflowJob" {
+				jobRunID, ok := apiEvent.Payload["ID"].(float64)
+				status, okStatus := apiEvent.Payload["Status"].(string)
+				if ok && okStatus && status == sdk.StatusWaiting.String() {
+					// wait for the grace time before pushing the job in the channel
+					go func() {
+						time.Sleep(time.Duration(graceTime) * time.Second)
+						job, err := c.QueueJobInfo(int64(jobRunID))
+
+						// Do not log the error if the job does not exist
+						if sdk.ErrorIs(err, sdk.ErrWorkflowNodeRunJobNotFound) {
+							return
+						}
 
-				if c.config.Verbose {
-					fmt.Println("Old Jobs Queue size: ", len(queue))
-				}
+						if err != nil {
+							errs <- fmt.Errorf("unable to get job %v info: %v", jobRunID, err)
+							return
+						}
 
-				shrinkQueue(&queue, cap(jobs))
-				for _, j := range queue {
-					jobs <- j
+						// push the job in the channel
+						if job.Status == sdk.StatusWaiting.String() && job.BookedBy.Name == "" {
+							jobs <- *job
+						}
+					}()
 				}
 			}
+
 		case <-jobsTicker.C:
 			if c.config.Verbose {
 				fmt.Println("jobsTicker")
 			}
 
 			if jobs != nil {
 				queue := sdk.WorkflowQueue{}
-
-				url, _ := url.Parse("/queue/workflows")
-				q := url.Query()
-				if ratioService != nil {
-					q.Add("ratioService", fmt.Sprintf("%d", *ratioService))
-				}
-				if modelType != "" {
-					q.Add("modelType", modelType)
-				}
-				url.RawQuery = q.Encode()
-				_, header, _, errReq := c.RequestJSON(http.MethodGet, url.String(), nil, &queue, SetHeader(RequestedIfModifiedSinceHeader, t0.Format(time.RFC1123)))
-				if errReq != nil {
-					errs <- sdk.WrapError(errReq, "Unable to load jobs")
-					continue
-				}
-
-				apiTimeHeader := header.Get(ResponseAPITimeHeader)
-
-				var errParse error
-				t0, errParse = time.Parse(time.RFC3339, apiTimeHeader)
-				if errParse != nil {
-					errs <- sdk.WrapError(errParse, "Unable to load jobs, failed to parse API Time")
-					continue
+				if _, err := c.GetJSON("/queue/workflows", &queue); err != nil {
```

Comment:
missing query params modelType & ratio:

```go
url, _ := url.Parse("/queue/workflows")
q := url.Query()	
if ratioService != nil {	
    q.Add("ratioService", fmt.Sprintf("%d", *ratioService))	
}	
if modelType != "" {	
    q.Add("modelType", modelType)	
}	
url.RawQuery = q.Encode()
_, header, _, errReq := c.RequestJSON(http.MethodGet, url.String(), nil, &queue, SetHeader(RequestedIfModifiedSinceHeader, t0.Format(time.RFC1123)))
```

On master branch, the oldJobsTicker does not do that, but it's a mistake. I don't fix it on master to avoid conflict with this branch. ie: hatchery swarm with ratioService 100% receives 'old' jobs without service, but it's not expected. The ticket jobsTicker is ok (no worker without service send to hatchery swarm)